gcs/src/gcs_group.cpp

```diff
@@ -1165,7 +1185,10 @@ group_find_ist_donor (const gcs_group_t* const group,
              (long long)ist_seqno, (long long)lowest_cached_seqno,
              (long long)conf_seqno, (long long)safe_ist_seqno);
 
-    if (ist_seqno < safe_ist_seqno) {
+    // We should ignore this heuristic if the request contains only a IST part,
+    // since otherwise we cannot fulfill the request:
+
+    if (ist_seqno < safe_ist_seqno && ! ist_only) {
```

Comment:
I am not suggesting to change the algorithm but instead tweak the check as follows:

Current:
    gcs_seqno_t safe_ist_seqno = lowest_cached_seqno + safety_gap;

Check based on ist
    gcs_seqno_t safe_ist_seqno = lowest_cached_seqno + ((ist_only) ? 0 : safety_gap);

This will ensure the debug message and if check can be left untouched.